go/test/endtoend/tabletmanager/tablet_health_test.go

```diff
@@ -186,6 +207,19 @@ func TestHealthCheck(t *testing.T) {
 
 	time.Sleep(tabletHealthcheckRefreshInterval)
 
+	// After replication has started, we sometimes see that MySQL reports a high
+	// replication lag but this clears really quick.
+	// In order to prevent the test from failing, we should keep checking with a timeout -
+	timeout = time.Now().Add(10 * time.Second)
```

Comment:
nit: this a pattern we might want to move to our test utils so we can reuse it

Reply:
Where do you want me to move this?

Reply:
Hmmm... That is in the `vtgate` package. Maybe we don't move it now, unless you can think of a good place for it to live in